### PR TITLE
9887 loan memo summary page

### PR DESCRIPTION
**Pull request by emitche**
- Display loan summary questions and groups in print loan memo page

### CODE REVIEW

File: app/models/question_set.rb [hunk 1]
```diff
@@ -78,6 +78,10 @@ def question(question_identifier, required: true)
     result
   end
 
+  def summary_questions
```
Comment (emitche) [inline app/models/question_set.rb]: @meltheadorable I don't like this particular method. If you have a suggested improvement here, please let me know. The code only needs to check if there are questions with the display_in_summary attribute set to `true`. The code is not using that list of questions to populate any content.

File: app/views/admin/loans/questionnaires/_questionnaire_toc.html.slim [hunk 2]
```diff
@@ -1,4 +1,6 @@
 ul
+  - if @response_set.question_set.summary_questions.length > 0
```
Comment (emitche) [inline app/views/admin/loans/questionnaires/_questionnaire_toc.html.slim]: @meltheadorable See earlier comment about refactoring `summary_questions`.

File: app/models/question_set.rb (hunk 1 above)
Comment (meltheadorable) [reply app/models/question_set.rb]: There may be another option here involving scopes, but if this is really only checking whether they exist, you should be able to do

```ruby
def summary_questions?
  questions.where(display_in_summary: true).count > 0
end
```

which should do a much smaller/faster query, and would avoid you needing to do the length check later in the view.

File: app/views/admin/loans/questionnaires/_questionnaire_toc.html.slim (hunk 2 above)
Comment (meltheadorable) [reply app/views/admin/loans/questionnaires/_questionnaire_toc.html.slim]: See other comment.

**Review by meltheadorable — APPROVED** [review]

File: app/models/question_set.rb (hunk 1 above)
Comment (emitche) [reply app/models/question_set.rb]: Thank you!

**Comment by emitche** [comment]
QA completed in 9886

**Pull request merged**